### ISSUE TITLE
Ajax defaulting to stringifying all bodies that are typeof === 'object'

### ISSUE THREAD

**Issue by blesh** [Reactive-Extensions/RxJS-DOM#44]
The problem here it's not always desirable to stringify anything that is of type "object". Blobs for example test positive for `typeof someBlob === 'object'`, but you can `xhr.send(someBlob)` directly.

In other words, the current implementation doesn't allow the developer to send: 

- ArrayBufferView
- Blob
- Document
- FormData

which are all legitimate arguments for `xhr.send(x);`

**Comment by mattpodwysocki** [comment]
@blesh then what do we do to propose fixing this?

**Comment by blesh** [comment]
Well, could either:

1) Break the API and force all bodies to be their final form (devs have to JSON.parse themselves)

Or

2) check instanceof on other supported types (Blob for example) before doing the JSON.parties. 

I'm on the fence with either approach.

3) Another option might be too give it a configurable serialization and deserialization parameters that default to what you currently have.

**Comment by blesh** [comment]
Sorry about the typos, I'm on my phone

**Comment by blesh** [comment]
So, I was up late watching a baby and I went overboard last night after talking to @jeffbcross about HTTP and RxJS... [I have a branch here](https://github.com/blesh/RxJS-DOM/tree/jsonp/src) in which I've played with refactoring both Ajax and JSONP implementations.

[ajax.js](https://github.com/blesh/RxJS-DOM/blob/jsonp/src/ajax.js)
[jsonp.js](https://github.com/blesh/RxJS-DOM/blob/jsonp/src/jsonp.js)

**What's in that branch is just concept code and is untested**. I also have a lot of questions about the existing JSONP implementation. In particular the recursed functionality, which I've not yet seen in a JSONP impl. 

I'm the least married to the ideas represented in my JSONP implementation, as I don't often use JSONP. The main idea was to unify the input to and the output from the JSONP and AJAX observables so they were more interchangeable.

- JSONP & AJAX implementations now have "normalized" output. This would be a breaking change for JSONP and (possible) breaking change for Ajax implementations. The goal here was to make a JSONP observable or Ajax observable more "swappable" meaning that they both emit data that has a similar structure.
- JSONP & AJAX impls no longer have to potential to mutate the "settings" object passed to them.
- JSONP no longer puts handlers in global scope.
- AJAX impl now supports an optional `progressObserver` parameter on the passed settings. This exposes XHR2 features, emitting ProgressEvents for `onprogress`, `onerror`, `onload` and `onabort`.
- AJAX impl now supports an optional `uploadProgress` parameter on the passed settings. This exposes XHR2 `upload` events in a similar manner to the progressObserver above.
- AJAX impl now checks `content-type` in `headers` passed for `application/json` along with `typeof` the `body` passed. If content type is `application/json` and the body passed is an `object`, it will JSON.parse() the object. The only time this might not be ideal is when, for some reason someone really wants to send a Blob or something with a content-type of application/json. That seems like an extreme edge case though.
- AJAX now supports sending objects like `Blob` and `FormData` in the body without attempting to JSON.parse() it.
